src/index.ts

```diff
@@ -15,15 +15,40 @@ import rehypeReact, { Options as RehypeReactOptions } from 'rehype-react';
 
 type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
 
-export interface UseRemarkOptions {
-  remarkParseOptions?: Partial<RemarkParseOptions>;
+export interface UseRemarkSyncOptions {
+  remarkParseOptions?: RemarkParseOptions;
   remarkToRehypeOptions?: RemarkRehypeOptions;
   rehypeReactOptions?: PartialBy<
     RehypeReactOptions<typeof createElement>,
     'createElement'
   >;
   remarkPlugins?: PluggableList;
   rehypePlugins?: PluggableList;
+}
+
+export const useRemarkSync = (
+  source: string,
+  {
+    remarkParseOptions,
+    remarkToRehypeOptions,
+    rehypeReactOptions,
+    remarkPlugins = [],
+    rehypePlugins = [],
+  }: UseRemarkOptions = {}
```

Comment:
@ChristianMurphy @wooorm should this be `UseRemarkSyncOptions` instead? (considering that `onError` isn't used here)

PR here:

-

Reply:
@karlhorky this PR is 3 years old.
Some of this code has evolved and been reworked since.
Happy to discuss in an issue or PR if you believe this is a present issue.